Enable insts to be loaded before server startup

pfeodrippe opened the pull request:
This makes `:group`, `:instance-group`, `:fx-group` and `:mixer` atoms in a `Inst` so we can enable the loading of the insts at anytime, even before the server startup.

plexus commented:
This sounds super useful. How confident are we that there's no existing user code touching those fields? I feel like at least this should get a big shoutout in the CHANGELOG that this is a breaking change. The alternative could be to turn Inst into a deftype and use `:volatile-mutable`, but that would be a bigger refactoring as you'd need to reimplement the map/record semantics... and it would still be  breaking in the sense that existing code may specifically rely on inst being a `record?`... What do you think?

pfeodrippe commented:
> This sounds super useful. How confident are we that there's no existing user code touching those fields? I feel like at least this should get a big shoutout in the CHANGELOG that this is a breaking change. The alternative could be to turn Inst into a deftype and use `:volatile-mutable`, but that would be a bigger refactoring as you'd need to reimplement the map/record semantics... and it would still be breaking in the sense that existing code may specifically rely on inst being a `record?`... What do you think?

Yeah, I was thinking about using Potemkin for this, wdyt? I have used it in some projects and it has been pretty good, at least I could see if it would work, otherwise, we would have to deftype it ourselves indeed.

pfeodrippe commented on src/overtone/studio/inst.clj:
```diff
@@ -14,24 +14,26 @@
+            [overtone.sc.machinery.server.comms :refer [with-server-sync]]
+            [overtone.libs.deps :as ov.deps]))
+
+(ov.lib/defrecord-ifn-2 Inst [name full-name params args sdef
```
No need for __RECORDS__ anymore as we have a simple caching using the form.

pfeodrippe commented:
@plexus Added potemkin and implemented an automatic deref for selected fields (you have to use the `^:deref` metadata as we can have other fields (e.g. `:pan`) that are atoms and which we don't want to do the deref automatically), things seems to be fine now, added a small breaking chang to the changelog that I guess we really don't to handle, let me know

pfeodrippe commented on src/overtone/studio/inst.clj:
```diff
@@ -200,6 +211,31 @@
 
+(defn- load-all-insts
+  []
+  ;; TODO Get atoms using something else.
```
This TODO is already done, will remove it before merging o/

pfeodrippe commented:
Can I merge this one?

plexus commented:
![image](https://github.com/user-attachments/assets/007c145d-8a2e-43d6-84b2-77006caa7ee6)

Nice one! Feel free to just merge stuff if you feel confident about it. We can always follow up with a future PR.

pfeodrippe commented:
Sure, thanks! Will do it next time, appreciate it ;D
